### PR TITLE
add details on blob storage encryption

### PR DESCRIPTION
**Pull request by p0pr0ck5**

### CODE REVIEW

**Review by nathanl79 — APPROVED** [review]
Well written!

**Review by sudomateo — APPROVED** [review]

File: website/docs/cloud-docs/architectural-details/data-security.mdx [hunk 1]
```diff
@@ -40,4 +40,23 @@ seriously. This table lists which parts of the Terraform Cloud and Terraform Ent
...
+which uses the default transit encryption cipher (AES-GCM with a 256-bit AES key
+and a 96-bit nonce), and stored alongside the object, in a pattern known as
+_envelope encryption_.
+
+Encryption key material is generated via the Vault transit secret engine's
+[datakey generation](https://www.vaultproject.io/api-docs/secret/transit#generate-data-key),
+using bit material from the kernel's cryptographically secure pseudo-random
+number generator (CSPRNG) as the `context` value. Because blob storage encryption
+generates a unique key for each object, and relies on envelope encryption,
+the encryption key material itself for any given object is not rotated.
```
Comment (laurapacilio) [inline website/docs/cloud-docs/architectural-details/data-security.mdx:62]:
```suggestion
The Vault transit secret engine's
[datakey generation](https://www.vaultproject.io/api-docs/secret/transit#generate-data-key) creates the encryption key material using bit material from the kernel's cryptographically secure pseudo-random
number generator (CSPRNG) as the `context` value. Blob storage encryption generates a unique key for each object and relies on envelope encryption, so Vault does not rotate the encryption key material for individual objects.
```

File: website/docs/cloud-docs/architectural-details/data-security.mdx [hunk 2]
```diff
@@ -40,4 +40,23 @@ seriously. This table lists which parts of the Terraform Cloud and Terraform Ent
...
+
+## Blob Storage Encryption
+
+All objects persisted to blob storage are symmetrically encrypted prior to being
+written. Each object is encrypted with a unique encryption key. Objects are
+encrypted using 128 bit AES in CTR mode. The key material itself is round-tripped
+through the [Vault transit secret engine](https://www.vaultproject.io/docs/secrets/transit/index.html),
+which uses the default transit encryption cipher (AES-GCM with a 256-bit AES key
+and a 96-bit nonce), and stored alongside the object, in a pattern known as
+_envelope encryption_.
```
Comment (laurapacilio) [inline website/docs/cloud-docs/architectural-details/data-security.mdx:55]:
```suggestion
All objects persisted to blob storage are symmetrically encrypted prior to being
written. Each object is encrypted with a unique encryption key. Objects are
encrypted using 128 bit AES in CTR mode. The key material is processed
through the [Vault transit secret engine](https://www.vaultproject.io/docs/secrets/transit/index.html),
which uses the default transit encryption cipher (AES-GCM with a 256-bit AES key
and a 96-bit nonce), and stored alongside the object. This pattern is called envelope encryption.
```

**Review by laurapacilio — APPROVED** [review]
Approving so I don't block! A couple of suggestions for flow. Please ignore if I've materially changed the meaning - I'm not super familiar with this subject matter. Thank you!

**Pull request merged**